jobserver/models/core.py

```diff
@@ -877,6 +892,19 @@ def clean(self):
         super().clean()
         self.email = self.__class__.objects.normalize_email(self.email)
 
+    @classmethod
+    @transaction.atomic()
+    def from_login_token(cls, token):
+        user = User.objects.get(
```

Comment:
So, I think we should add the username/id into the url also, and look up based on both username *and* token.  This prevents link from ever logging you in as the wrong user, if for some reason the two tokens where the same.  Unlikely, I know, but easily preventable so we probably should do it.

Also, we may want to distinguish between token expired and invalid token, possibly. So it might be worth separating the expiry check out from the get query.

I think we'd want to log whether it is one or other, even if what we say to the user is the same in both cases, to help us debug auth failures more easily.

Also, if we do the expiry check separately, and it has expired, we may want to reset it in the db right now, just to be clean and safe.

If we do respond to the user differently depending on reason for failure, then its more helpful to the user, but arguably leaks information about the token's existence or not.